crates/uv/src/commands/project/run.rs

```diff
@@ -994,9 +996,28 @@ pub(crate) async fn run(
     // Ignore signals in the parent process, deferring them to the child. This is safe as long as
     // the command is the last thing that runs in this process; otherwise, we'd need to restore the
     // signal handlers after the command completes.
-    let _handler = tokio::spawn(async { while tokio::signal::ctrl_c().await.is_ok() {} });
+    let mut term_signal = signal(SignalKind::terminate())?;
+    let mut int_signal = signal(SignalKind::interrupt())?;
 
-    let status = handle.wait().await.context("Child process disappeared")?;
+    let status = select! {
+        status = handle.wait() => status,
+
+        // `SIGTERM`
+        _ = term_signal.recv() => {
+            handle.kill().await?;
+            handle.wait().await.context("Child process disappeared")?;
+            return Ok(ExitStatus::Failure);
+        }
+
+        // `SIGINT`
+        _ = int_signal.recv() => {
+            handle.kill().await?;
+            handle.wait().await.context("Child process disappeared")?;
+            return Ok(ExitStatus::Failure);
```

Comment:
`uv run` should preferably preserve the exit code from the child, not assume this was an error exit. (I'm sorry for more comments if they are superfluous, but it's better to get the issues out in the sun.)